disnake/onboarding.py

```diff
@@ -0,0 +1,201 @@
+        ]
+        self.enabled: bool = data["enabled"]
+        self.default_channel_ids: FrozenSet[int] = (
+            frozenset(map(int, exempt_channels))
+            if (exempt_channels := data["default_channel_ids"])
+            else frozenset()
+        )
+
+    def __repr__(self) -> str:
+        return (
+            f"<Onboarding guild={self.guild!r} prompts={self.prompts!r} enabled={self.enabled!r}>"
+        )
+
+    @property
+    def default_channels(self) -> List[GuildChannel]:
+        """List[:class:`abc.GuildChannel`]: The list of channels that will be shown to new members by default."""
+        return list(filter(None, map(self.guild.get_channel, self.default_channel_ids)))
+
+
+class OnboardingPrompt(Hashable):
+    """Represents an onboarding prompt.
+
+    .. versionadded:: 2.9
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The onboarding prompt's ID.
+    type: :class:`OnboardingPromptType`
+        The onboarding prompt's type.
+    options: List[:class:`OnboardingPromptOption`]
+        The onboarding prompt's options.
+    title: :class:`str`
+        The onboarding prompt's title.
+    single_select: :class:`bool`
+        Whether users are limited to selecting one option for the prompt.
+    required: :class:`bool`
+        Whether the prompt is required before a user completes the onboarding flow.
+    in_onboarding: :class:`bool`
+        Whether the prompt is present in the onboarding flow.
+        If ``False``, the prompt will only appear in community customization.
+    """
+
+    __slots__ = (
+        "guild",
+        "id",
+        "title",
+        "options",
+        "single_select",
+        "required",
+        "in_onboarding",
+        "type",
+    )
+
+    def __init__(self, *, guild: Guild, data: OnboardingPromptPayload) -> None:
+        self.guild = guild
+
+        self.id: int = int(data["id"])
+        self.title: str = data["title"]
+        self.options: List[OnboardingPromptOption] = [
+            OnboardingPromptOption(data=option, guild=guild) for option in data["options"]
+        ]
+        self.single_select: bool = data["single_select"]
+        self.required: bool = data["required"]
+        self.in_onboarding: bool = data["in_onboarding"]
+        self.type: OnboardingPromptType = try_enum(OnboardingPromptType, data["type"])
+
+    def __str__(self) -> str:
+        return self.title
+
+    def __repr__(self) -> str:
+        return (
+            f"<OnboardingPrompt id={self.id!r} title={self.title!r} options={self.options!r}"
+            f" single_select={self.single_select!r} required={self.required!r}"
+            f" in_onboarding={self.in_onboarding!r} type={self.type!r}>"
+        )
+
+
+class OnboardingPromptOption(Hashable):
+    """Represents an onboarding prompt option.
+
+    .. versionadded:: 2.9
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The prompt option's ID.
+    emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`]]
```

Comment:
```suggestion
    emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`, :class:`str`]]
```
(`get_reaction_emoji` typings are slightly incorrect, fixed in #980)